src/tasks/run_closed_loop.py

```diff
@@ -1,5 +1,45 @@
 """Run all components of the BCI closed loop."""
+import logging
+import os
 import subprocess
+import tempfile
+
+from neural_data_simulator.settings import LogLevel
+from neural_data_simulator.util.runtime import configure_logger
+from neural_data_simulator.util.runtime import initialize_logger
+
+SCRIPT_NAME = "nds-run-closed-loop"
+logger = logging.getLogger(__name__)
+
+
+def _terminate_process(label: str, popen_process: subprocess.Popen, timeout: int = 5):
+    logger.info(f"Terminating {label}")
+    popen_process.terminate()
+    try:
+        popen_process.wait(timeout)
+    except subprocess.TimeoutExpired:
+        logger.warning(f"{label} did not terminate in time. Killing it")
+        popen_process.kill()
+
+
+def _create_pipe_file(pipe_name: str):
+    temp_dir = tempfile.mkdtemp()
+    pipe_path = os.path.join(temp_dir, pipe_name)
+    os.mkfifo(pipe_path)
+    return (temp_dir, pipe_path)
+
+
+def _wait_for_main_task_finish(pipe_path: str):
```

Comment:
please add the return type annotation and a short docstring explaining what it does, describing what script writes to the pipe, also the name of the function is hiding one important aspect of it: the True/False distinction of the return value.
My suggestion is to move the try/except block to the calling function and make _wait_for_main_task_finish return None

Reply:
The try/except block inside this function removes the complexity in the calling function that already has a lot of code. Let me rewrite it a little bit, and then you check if it's good. :)

Reply:
Please, take a second look, I've sent the changes :)

Reply:
The problem here is that a function called `wait_...` is doing something else than "waiting" as it returns a boolean, I will come up with a code snippet with a refactoring suggestion

Reply:
I removed this method because a new condition was needed in the loop. It would require at least three parameters for a simple loop; probably not worth it. 🤔